packages/wallets/src/components/WalletsResetMT5Password/WalletsResetMT5Password.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from 'react';
+import { useTradingPlatformPasswordReset } from '@deriv/api';
+import { PlatformDetails } from '../../features/cfd/constants';
+import useDevice from '../../hooks/useDevice';
+import { TPlatforms } from '../../types';
+import { validPassword } from '../../utils/passwordUtils';
+import { ModalStepWrapper, WalletButton, WalletButtonGroup, WalletPasswordField, WalletText } from '../Base';
+import { WalletPasswordFieldProps } from '../Base/WalletPasswordField/WalletPasswordField';
+import { useModal } from '../ModalProvider';
+import WalletSuccessResetMT5Password from './WalletSuccessResetMT5Password';
+import './WalletsResetMT5Password.scss';
+
+type WalletsResetMT5PasswordProps = {
+    onChange: WalletPasswordFieldProps['onChange'];
+    password: WalletPasswordFieldProps['password'];
+    platform: Exclude<TPlatforms.All, 'ctrader'>;
+    verificationCode: string;
+};
+
+const WalletsResetMT5Password = ({ onChange, password, platform, verificationCode }: WalletsResetMT5PasswordProps) => {
+    const { title } = PlatformDetails[platform];
+    const { mutate, status } = useTradingPlatformPasswordReset();
+    const { hide, show } = useModal();
+    const { isDesktop, isMobile } = useDevice();
+
+    const handleSubmit = () => {
+        mutate({
+            new_password: password,
+            platform,
+            verification_code: verificationCode,
+        });
+    };
+
+    useEffect(() => {
+        if (status === 'success') {
+            localStorage.removeItem(`verification_code.trading_platform_${platform}_password_reset`);
```

Comment:
I don't think we should use local storage at all to store verification codes,

Reply:
I'm not sure if we should let old patterns go through (especially the ones which generates tight coupling between legacy bits - and that localStorage support is within legacy area which ideally we should cut off). 
Though I also understand it might require more time to implement properly so might not be ideal moment for that. Happy to help with that if needed. 

Any thoughts @markwylde-deriv @yashim-deriv ?

Reply:
It looks like this PR isn't what's actually setting that verification code? So I'm presuming the email redirects (maybe to oauth first, but eventually) to a URL that needs to stay the same for none migrated clients, but then go here for the ones who have migrated?

Didn't we look at keeping this in the URL and passing it to the correct page. As in, capturing it at the landing page, and then keeping it in localStorage (if appstore/cashier) or pass it to wallets in the URL.

Reply:
there is generic code-setting logic within one of the stores / old packages, which sets local storage and arranges redirects, 
some bits of Wallets do not use local storage for that purpose, they handle verification code fully through url's

Reply:
I'm working with the similar PR (reset mt5 investor password) and really need to know wha't the final decision :) In my opinion it's not so hard to pass the code in URI param just for wallets package and doesn't touch legacy code, but we need to confirm from @yashim-deriv , do we need to make this change in this PR or later. Also, if we change a lot in core then this PR will not be merged and needs a QA attention. In this case I have a question: "Do we need to do this in this PR or later"? cc: @wojciech-deriv @markwylde-deriv @yashim-deriv @mahdiyeh-deriv what do you think?

Reply:
I do agree with you @aizad-deriv , better not to refactor this here since we are not decided it yet, I'll add a placeholder for this to pick it up later.